worlds/sm64ex/Rules.py

```diff
@@ -38,48 +38,59 @@ def set_rules(world, player: int, area_connections):
 
     #Special Rules for some Locations
     add_rule(world.get_location("Tower of the Wing Cap Switch", player), lambda state: state.has("Power Star", player, 10))
-    add_rule(world.get_location("Cavern of the Metal Cap Switch", player), lambda state: state.can_reach("Hazy Maze Cave",'Region',player))
+    add_rule(world.get_location("Cavern of the Metal Cap Switch", player), lambda state: state.can_reach("Hazy Maze Cave", 'Region', player))
     add_rule(world.get_location("Vanish Cap Under the Moat Switch", player), lambda state: state.can_reach("Basement", 'Region', player))
 
+    add_rule(world.get_location("BoB: Mario Wings to the Sky", player), lambda state: state.has("Cannon Unlock BoB", player))
     add_rule(world.get_location("BBH: Eye to Eye in the Secret Room", player), lambda state: state.has("Vanish Cap", player))
-    add_rule(world.get_location("DDD: Collect the Caps...", player), lambda state: state.has("Metal Cap", player) and
-                                                                              state.has("Vanish Cap", player))
-    add_rule(world.get_location("DDD: Pole-Jumping for Red Coins", player), lambda state: state.can_reach("Bowser in the Fire Sea",'Region',player))
+    add_rule(world.get_location("DDD: Collect the Caps...", player), lambda state: state.has("Vanish Cap", player))
+    add_rule(world.get_location("DDD: Pole-Jumping for Red Coins", player), lambda state: state.can_reach("Bowser in the Fire Sea", 'Region', player))
     add_rule(world.get_location("SL: Into the Igloo", player), lambda state: state.has("Vanish Cap", player))
     add_rule(world.get_location("WDW: Quick Race Through Downtown!", player), lambda state: state.has("Vanish Cap", player))
-    if (world.StrictCapRequirements[player].value):
+    add_rule(world.get_location("RR: Somewhere Over the Rainbow", player), lambda state: state.has("Cannon Unlock RR", player))
+
+    if world.AreaRandomizer[player]:
```

Comment:
Why remove the `.value`?

Reply:
It's not necessary to have `.value` on these because the `Option` class implements `__bool__`, so it will do this conversion implicitly. It's cleaner to just check the option directly.

Reply:
I'd rather you keep it then, for consistency among option access

Reply:
Consistency in which sphere? this PR makes it consistent within SM64 and this way of doing it is the consistent way in most AP worlds. Do you sill want this changed or is the PR good to merge as-is?

Reply:
If it's the way the rest of the codebase does it then I have no issues with it. What I meant was to make access to options the same regardless of option type, which I personally prefer.